lgc/patch/WaveSizeAdjust.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Trade secret of Advanced Micro Devices, Inc.
+ *  Copyright (c) 2019, Advanced Micro Devices, Inc., (unpublished)
+ *
+ *  All rights reserved. This notice is intended as a precaution against inadvertent publication and does not imply
+ *  publication or any waiver of confidentiality. The year included in the foregoing notice is the year of creation of
+ *  the work.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  WaveSizeAdjust.cpp
+ * @brief LLPC source file: WaveSizeAdjust pass
+ ***********************************************************************************************************************
+ */
+#include "lgc/patch/Patch.h"
+#include "lgc/state/PipelineShaders.h"
+#include "lgc/state/PipelineState.h"
+#include "llvm/Pass.h"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "lgc-wave-size-adjuster"
+
+using namespace lgc;
+using namespace llvm;
+
+namespace {
+
+// =====================================================================================================================
+// Pass to adjust wave size per shader stage heuristically.
+class WaveSizeAdjust final : public ModulePass {
```

Comment:
According to the naming, change it to PatchWaveSizeAdjust both for the class and the file.

Reply:
I thought the classes derived from Patch class have a prefix Patch for the class name and file name. Did I get the wrong meaning?

Reply:
Done.